documentation/devonfw-ide-contribution-getting-started.asciidoc

```diff
@@ -0,0 +1,160 @@
+Sometimes it needs more patience to get an acceptable result or to find out that a command you already knew includes what you need, if you add the proper tag to its execution.
+Take the commands and some of their tags mentioned in https://www.freecodecamp.org/news/linux-command-line-Bash-tutorial/[this rather brief tutorial] as an example. Also definetely try to use the help function from time to time, e.g. type in `ls --help` and you'll see how much more the command `ls` can offer. 
+On the other side, especialy at the beginning, it is better to get an overview of available tags by googling e.g. 
+*ls tags Bash* to get a way prettier overview plus examples regarding a commands tags like https://www.tecmint.com/15-basic-ls-command-examples-in-linux/[here] for `ls`.
+
+**If you have some experience with the Bash or directly want to see more:**
+
+The meaning of a tag or a sign combination (like `$*`, `$#`, or `$()`) can be hard to guess and also less easy to find during a google-search, because google won't directly search for signs.
+Then a syntax cheatsheet https://www.pcwdld.com/bash-cheat-sheet[like this] or https://devhints.io/bash[like that one] can spare you some searching time. 
+If you don't know what a positional parameter like `$1` is in Bash, then have a look at https://wiki.bash-hackers.org/scripting/posparams[this introduction] and play around with the tools mentioned to get a better understanding of how they work.
+It's definitively worth knowing the basics about positional parameters if you want to make some more advanced Bash-scripts.
+There are a few more topics that I want to mention to you as you may come across them regularly in Bash-scripting. 
+The first one concerns regular expressions, which you will find during various coding situations.
+Therefore have a look at https://tldp.org/LDP/abs/html/x17129.html[this rather short introdcution on regex] if you are unfamiliar with the so called regex. 
+Another one is about if-statements and square brackets for testing expressions by comparing them in different ways.
+Maybe you'll find something new or helpful https://ryanstutorials.net/bash-scripting-tutorial/bash-if-statements.php[here]. 
+Finally, to possibly increase your knowledge while scripting bash-files, try out https://www.shellcheck.net/[this spellchecker] that can give you many details on how to improve your code.
+
+**Directly regarding our project:**
+
+In case you want to get your hands dirty with code that is already in use and comes from our ide-project, then you can try out the functions written in our https://github.com/devonfw/ide/blob/master/scripts/src/main/resources/scripts/functions[functions-file]. 
+There are also some basic exercises related to our project that are not official yet, but you can ask for if you want to try some of them out to learn and challenge yourself a bit.
+
+== Documentation
+https://devonfw.com/website/pages/docs/devonfw-guide_ide.wiki_devonfw-ide.asciidoc.html
```

Comment:
FYI: I never put detailed links on `devonfw.com` from github. These links have never been stable. The link will be broken within weeks, months or at least after one year.
The entire devonfw.com website is continuously changing and another big redesign in already planned.
I read documentation on github where links are stable.